ansible_collections/arista/cvp/plugins/module_utils/device_tools.py

```diff
@@ -1739,6 +1780,106 @@ def reset_device(self, user_inventory: DeviceInventory):
+                            )
+                        )
+                        resp = self.__cv_client.api.validate_config_for_device(
+                            device_mac=device.system_mac,
+                            config=vc_configlet["config"],
+                        )
+                        MODULE_LOGGER.debug("resp is: {0}".format(str(resp)))
+                    except CvpApiError:
+                        MODULE_LOGGER.critical(
+                            "Error validation failed on device {0}".format(device.fqdn)
+                        )
+                        self.__ansible.fail_json(
+                            msg="Error validation failed on device {0}".format(
+                                device.fqdn
+                            )
+                        )
+                    else:
+                        if "result" in resp:
+                            result_data.changed = True
+                            result_data.success = True
+                        if "warnings" in resp and resp["warningCount"] > 0:
+                            err_msg = resp["warnings"]
+                            msg = f"Configlet validation failed with {err_msg}"
+                            result_data.success = False
+                            result_data.count += 1
+                            MODULE_LOGGER.error(msg)
+                            device_data["warnings"].append(
+                                {"device": device.hostname, "warnings": err_msg}
+                            )
+                        if "errors" in resp:
+                            err_msg = resp["errors"]
+                            msg = f"Configlet validation failed with {err_msg}"
+                            result_data.success = False
+                            result_data.count += 1
+                            MODULE_LOGGER.error(msg)
+                            device_data["errors"].append(
+                                {"device": device.hostname, "errors": err_msg}
+                            )
+                result_data.add_entry(
+                    "{0} validated against {1}".format(
+                        device.info[self.__search_by], vc_configlet
+                    )
+                )
+            results.append(result_data)
+        MODULE_LOGGER.debug("device_data is: {0}".format(str(device_data)))
+        if len(device_data["errors"]) > 0 and validate_mode in [
+            ModuleOptionValues.VALIDATE_MODE_STOP_ON_WARNING,
+            ModuleOptionValues.VALIDATE_MODE_STOP_ON_ERROR,
+        ]:
+            self.__ansible.fail_json(msg=str(device_data))
+        elif (
+            len(device_data["warnings"]) > 0
+            and validate_mode == ModuleOptionValues.VALIDATE_MODE_STOP_ON_WARNING
+        ):
+            self.__ansible.fail_json(msg=str(device_data))
+        else:
+            self.__ansible.exit_json(msg=str(device_data))
+        return results
```

Comment:
This line is unreachable. If there are no errors and warning, then we should return `results`